rskj-core/src/main/java/co/rsk/rpc/modules/eth/EthModuleTransactionInstant.java

```diff
@@ -0,0 +1,121 @@
+
+public class EthModuleTransactionInstant implements EthModuleTransaction {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger("web3");
+
+    private final RskSystemProperties config;
+    private final Wallet wallet;
+    private final TransactionPool transactionPool;
+    private final MinerServer minerServer;
+    private final MinerClient minerClient;
+    private final Blockchain blockchain;
+
+    public EthModuleTransactionInstant(RskSystemProperties config, Wallet wallet, TransactionPool transactionPool, MinerServer minerServer, MinerClient minerClient, Blockchain blockchain) {
+        this.config = config;
+        this.wallet = wallet;
+        this.transactionPool = transactionPool;
+        this.minerServer = minerServer;
+        this.minerClient = minerClient;
+        this.blockchain = blockchain;
+    }
+
+    @Override
+    public synchronized String sendTransaction(Web3.CallArguments args) {
+        Account account = this.wallet.getAccount(new RskAddress(args.from));
+        String s = null;
+        try {
+            String toAddress = args.to != null ? Hex.toHexString(stringHexToByteArray(args.to)) : null;
+
+            BigInteger value = args.value != null ? TypeConverter.stringNumberAsBigInt(args.value) : BigInteger.ZERO;
+            BigInteger gasPrice = args.gasPrice != null ? TypeConverter.stringNumberAsBigInt(args.gasPrice) : BigInteger.ZERO;
+            BigInteger gasLimit = args.gas != null ? TypeConverter.stringNumberAsBigInt(args.gas) : BigInteger.valueOf(GasCost.TRANSACTION_DEFAULT);
+
+            if (args.data != null && args.data.startsWith("0x")) {
+                args.data = args.data.substring(2);
+            }
+
+            Transaction tx;
+
+            synchronized (transactionPool) {
+                BigInteger accountNonce = args.nonce != null ? TypeConverter.stringNumberAsBigInt(args.nonce) : transactionPool.getPendingState().getNonce(account.getAddress());
+                tx = Transaction.create(config, toAddress, value, accountNonce, gasPrice, gasLimit, args.data);
+                tx.sign(account.getEcKey().getPrivKeyBytes());
+                transactionPool.addTransaction(tx.toImmutableTransaction());
+                s = tx.getHash().toJsonString();
+            }
+
+            mineTransaction();
+
+            return s;
+        } finally {
+            LOGGER.debug("eth_sendTransaction({}): {}", args, s);
+        }
+    }
+
+    @Override
+    public String sendRawTransaction(String rawData) {
+        String s = null;
+        try {
+            Transaction tx = new ImmutableTransaction(stringHexToByteArray(rawData));
+
+            if (null == tx.getGasLimit()
+                    || null == tx.getGasPrice()
+                    || null == tx.getValue()) {
+                throw new JsonRpcInvalidParamException("Missing parameter, gasPrice, gas or value");
+            }
+
+            transactionPool.addTransaction(tx);
+
+            mineTransaction();
```

Comment:
same coment for sendRawTransaction, try moving the logic not another method

Reply:
same here

Reply:
ok